compiler/qsc_rca/src/lib.rs

```diff
@@ -0,0 +1,559 @@
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let mut indent = set_indentation(indented(f), 0);
+        write!(indent, "CallableComputeProperties:",)?;
+        indent = set_indentation(indent, 1);
+        write!(indent, "\nbody: {}", self.body)?;
+        match &self.adj {
+            Some(spec) => write!(indent, "\nadj: {spec}")?,
+            None => write!(indent, "\nadj: <none>")?,
+        }
+        match &self.ctl {
+            Some(spec) => write!(indent, "\nctl: {spec}")?,
+            None => write!(indent, "\nctl: <none>")?,
+        }
+        match &self.ctl_adj {
+            Some(spec) => write!(indent, "\nctl-adj: {spec}")?,
+            None => write!(indent, "\nctl-adj: <none>")?,
+        }
+        Ok(())
+    }
+}
+
+/// The compute properties associated to a callable or one of its elements.
+#[derive(Clone, Debug)]
+pub struct ApplicationsTable {
+    /// The inherent compute kind when all parameters are bound to static values.
+    pub inherent: ComputeKind,
+    /// Each element in the vector represents the compute kind of a call application when the parameter associated
+    /// to the vector index is bound to a dynamic value.
+    pub dynamic_param_applications: Vec<ComputeKind>,
+}
+
+impl ApplicationsTable {
+    pub fn new(params_count: usize) -> Self {
+        Self {
+            inherent: ComputeKind::Classical,
+            dynamic_param_applications: vec![ComputeKind::Classical; params_count],
+        }
+    }
+}
+
+impl Display for ApplicationsTable {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let mut indent = set_indentation(indented(f), 0);
+        write!(indent, "ApplicationsTable:",)?;
+        indent = set_indentation(indent, 1);
+        write!(indent, "\ninherent: {}", self.inherent)?;
+        write!(indent, "\ndynamic_param_applications:")?;
+        if self.dynamic_param_applications.is_empty() {
+            write!(indent, " <empty>")?;
+        } else {
+            indent = set_indentation(indent, 2);
+            for (para_index, param_compute_kind) in
+                self.dynamic_param_applications.iter().enumerate()
+            {
+                write!(indent, "\n[{}]: {}", para_index, param_compute_kind)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl ApplicationsTable {
+    pub fn derive_application_compute_kind(
+        &self,
+        input_params_dynamism: &Vec<bool>,
+    ) -> ComputeKind {
+        assert!(self.dynamic_param_applications.len() == input_params_dynamism.len());
+        let mut compute_kind = self.inherent.clone();
+        for (is_dynamic, dynamic_param_application) in input_params_dynamism
+            .iter()
+            .zip(self.dynamic_param_applications.iter())
+        {
+            if *is_dynamic {
+                compute_kind = aggregate_compute_kind(compute_kind, dynamic_param_application);
+            }
+        }
+        compute_kind
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum ComputeKind {
```

Comment:
`ComputeKind` (and its components) should implement `Copy`

Reply:
Fixed.